Podcast/Cache.swift

```diff
@@ -0,0 +1,107 @@
+//  Copyright © 2017 Cornell App Development. All rights reserved.
+//
+
+import UIKit
+import SwiftyJSON
+
+class Cache: NSObject {
+    static let sharedInstance = Cache()
+    
+    // NOTE: We might want to thing about limiting cache size in the future,
+    // but not something to worry about now. 
+    private var episodeCache: [String: Episode]!
+    private var seriesCache: [String: Series]!
+    private var userCache: [String: User]!
+    
+    private override init() {
+        episodeCache = [:] // TODO: in future maybe store cache!
+        seriesCache = [:]
+        userCache = [:]
+    }
+    
+    func reset() {
+        // TODO: this probably shouldn't be called. Gotta test what happens to views.
+        // Possible error: the cache is reset, but views retain references to objects,
+        // thus causing errors when cache is refilled. Views not flushed will be out
+        // of sync with episodes loaded into reset cache from another view's endpoint
+        // call. 
+        episodeCache = [:]
+    }
+    
+    // Takes in episode JSON and adds/updates cache
+    // Adds new if not present, updates cache if already present
+    // ONLY should be called from endpoint requests!!!
+    func update(episodeJson: JSON) -> Episode {
+        let id = episodeJson["id"].stringValue
+        if let episode = episodeCache[id] {
+            // Update current episode object to maintain living object
+            episode.title = episodeJson["title"].stringValue
+            episode.descriptionText = episodeJson["summary"].stringValue
+            episode.isRecommended = episodeJson["is_recommended"].boolValue
+            episode.isBookmarked = episodeJson["is_bookmarked"].boolValue
+            episode.numberOfRecommendations = episodeJson["recommendations_count"].intValue
+            episode.seriesTitle = episodeJson["series"]["title"].stringValue
+            episode.seriesID = episodeJson["series"]["id"].stringValue
+            episode.duration = episodeJson["duration"].stringValue
+            episode.tags = episodeJson["tags"].stringValue.components(separatedBy: ";").map({ tag in Tag(name: tag) })
+            episode.audioURL = URL(string: episodeJson["audio_url"].stringValue)
+            episode.dateCreated = DateFormatter.parsingDateFormatter.date(from: episodeJson["pub_date"].stringValue) ?? Date()
+            episode.smallArtworkImageURL = URL(string: episodeJson["series"]["image_url_sm"].stringValue)
+            episode.largeArtworkImageURL = URL(string: episodeJson["series"]["image_url_lg"].stringValue)
+            return episode
+        } else {
+            let episode = Episode(json: episodeJson)
+            episodeCache[id] = episode
+            return episode
+        }
+    }
+    
+    func update(seriesJson: JSON) -> Series {
+        let id = seriesJson["id"].stringValue
+        if let series = seriesCache[id] {
+            series.title = seriesJson["title"].stringValue
```

Comment:
same thing here with moving the json updating logic to a series model with a function call update(withJson json: Json)

Reply:
also have one `return series`

Reply:
They're two different objects though again